jetty-client/src/main/java/org/eclipse/jetty/client/HttpReceiver.java

```diff
@@ -477,6 +481,9 @@ protected void reset()
      */
     protected void dispose()
     {
+        if (decoder instanceof Destroyable) {
+            ((Destroyable) decoder).destroy();
+        }
```

Comment:
Please refactor the destruction of the decoder into a method, say `destroyDecoder(ContentDecoder)`, and call this new method from both `reset()` and `dispose()`